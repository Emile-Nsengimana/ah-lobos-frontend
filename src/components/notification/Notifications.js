/* eslint-disable react/jsx-no-target-blank */
import React from "react";
import propTypes from "prop-types";
import moment from "moment";
import Img from "../../assets/images/user.png";

const Notifications = props => {
  const inApp = [];
  if (props.notification !== undefined) {
    inApp.push(props.notification.filter(type => type.preference === "inApp"));
  }
  return (
    <>
      {inApp[0] !== undefined
        ? inApp[0].map(notify => (
            <a
              className="notification"
              key={notify.id}
              href={notify.url}
              onClick={() => {
                props.readUserNotification(notify.id);
              }}
            >
              <div className="notification-details">
                <div className="notification-img">
                  <img src={Img} alt="" />
                </div>
                <div className="notification-info">
                  <div className="notification-msg">{notify.message}</div>
                  <div className="notification-time">
                    {moment(moment(notify.createdAt).format("LLL")).fromNow()}
                  </div>
                </div>
              </div>
            </a>
          ))
        : null}
    </>
  );
};

export default Notifications;

Notifications.propTypes = {
  notification: propTypes.any,
  readUserNotification: propTypes.func
};
