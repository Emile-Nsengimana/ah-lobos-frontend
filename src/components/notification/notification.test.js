import React from "react";
import { shallow } from "enzyme";
import configureStore from "redux-mock-store";
import Notifications from "./Notifications";

const mockStore = configureStore();
const store = mockStore({});

const newProps = {
  readUserNotification: jest.fn(1)
};
const newcomponent = shallow(<Notifications {...newProps} store={store} />);
describe("get notification component", () => {
  it("renders without errors", () => {
    expect(newcomponent).toMatchSnapshot();
  });
});
