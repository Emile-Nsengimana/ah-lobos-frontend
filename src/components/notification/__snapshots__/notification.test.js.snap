// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get notification component renders without errors 1`] = `ShallowWrapper {}`;
